src/tool/hpcrun/audit/auditor.c

```diff
@@ -237,11 +240,22 @@ static void hook_open(uintptr_t* cookie, struct link_map* map) {
       }
     } else abort();
   }
-  entry->start = (void*)map->l_addr + start;
-  entry->end = (void*)map->l_addr + end;
+
+  if (is_vdso) {
+    entry->start = (void*)map->l_addr;
+    entry->end = (void*)map->l_addr + (end - start);
+  } else {
+    entry->start = (void*)map->l_addr + start;
+    entry->end = (void*)map->l_addr + end;
+  }
 
   // Since we don't use dl_iterate_phdr, we have to reconsitute its data.
-  entry->dl_info.dlpi_addr = (ElfW(Addr))map->l_addr;
+  if (is_vdso) {
+    entry->dl_info.dlpi_addr = (ElfW(Addr))(map->l_addr - start);
```

Comment:
Please modify `map->l_addr` here to the proper value, I believe it and `dl_info.dlpi_addr` are supposed to be in sync. Also that way I don't have to remember which is the "right" one when calculating offsets.

Reply:
resolved.